### ISSUE TITLE
empty e-mails incoming

### ISSUE THREAD

**Issue by timurb** [bosun-monitor/bosun#1453]
Today I've had an OpenTSDB outage and Bosun as expected emailed me a dozen of "Unknown" alerts.

What's wrong with that is soon I've started to messages with no subject and empty body while all of templates I defined in config file do have both template and body:
```
Date: Sun, 8 Nov 2015 07:40:27 +0000
Mime-Version: 1.0
X-Bosun-Server: tsdb
Content-Type: multipart/mixed;
 boundary=435514121d807855059eaa73cdea71ccbaa92a293f0eaafb0b48d9c5ec2f
To: ops@mydomain.com
From: ops@mydomain.com
Message-ID: <00000150e60a025a-e6d749cc-1edb-4844-93a6-9307544980d4-000000@us-west-2.amazonses.com>
X-SES-Outgoing: 2015.11.08-54.240.27.24
Feedback-ID: 1.us-west-2.LQSZzAmtlzVGV4VEOTpAGIGbe5mPLH9mf6V5OR2giLE=:AmazonSES
X-Original-Sender: ops@mydomain.com
X-Original-Authentication-Results: mx.google.com;       spf=pass (google.com:
 domain of 00000150e60a025a-e6d749cc-1edb-4844-93a6-9307544980d4-000000@us-west-2.amazonses.com
 designates 54.240.27.24 as permitted sender) smtp.mailfrom=00000150e60a025a-e6d749cc-1edb-4844-93a6-9307544980d4-000000@us-west-2.amazonses.com;
       dkim=pass header.i=@amazonses.com
Precedence: list
Mailing-list: list ops@mydomain.com; contact ops+owners@mydomain.com
List-ID: <ops.mydomain.com>
X-Spam-Checked-In-Group: ops@mydomain.com
X-Google-Group-Id: 612457435205
List-Help: <http://support.google.com/a/mydomain.com/bin/topic.py&topic=25838>, <mailto:ops+help@mydomain.com>

--435514121d807855059eaa73cdea71ccbaa92a293f0eaafb0b48d9c5ec2f

--435514121d807855059eaa73cdea71ccbaa92a293f0eaafb0b48d9c5ec2f--
```

I'm running Bosun 0.4.0 on Ubuntu 12.04, gathering incoming data via `/api/index` (sending to OpenTSDB in parallel) and sending out e-mails via AWS SES.

**Comment by timurb** [comment]
Copying here yesterday's slack chat for reference:
```
timurb [10:01 PM] 
I have the following lines when I see empty messages:
2015-11-10_18:45:17.05809 2015/11/10 18:45:17 info: notify.go:122: Batching and sending unknown notifications
2015-11-10_18:45:17.05810 2015/11/10 18:45:17 info: notify.go:152: Done sending unknown notifications


timurb [10:01 PM]
nothing besides these ones

timurb [10:01 PM]
but the same lines appear for valid messages too

captncraig [10:03 PM] 
nothing on 222,227, or 193 though?

timurb [10:05 PM] 
no, I see the messages from notify.go only for these 2 lines

captncraig [10:07 PM] 
We are still getting these empty emails about twice a day

captncraig [10:08 PM]
so I need to figure out where they are coming from

captncraig [10:08 PM]
I almost want to put a unique id in an smtp header anytime we send email and log a stack trace when we generate it

timurb [10:09 PM] 
in my case I think it was related to errors while accessing OpenTSDB

timurb [10:09 PM]
in both cases (weekend and today) I saw empty messages when my HBase cluster got down (edited)

captncraig [10:09 PM] 
interesting

captncraig [10:09 PM]
I think the errors should go into a different place now

captncraig [10:10 PM]
the little numbers on the top of the dashboard show errors.

captncraig [10:10 PM]
but things would also go unknown

captncraig [10:10 PM]
which is our best guess at this point

timurb [10:12 PM] 
yep, I can’t say anything about weekend but today I’ve seen one error in top right and received an empty message. Not sure what was the timing. I’ve shut down Bosun quickly to avoid receiving tons of them so I can’t say more than that for now.

timurb [10:13 PM]
I was checking OpenTSDB from cmdline using `check_tsd` and it simply produced timeouts


```

**Comment by captncraig** [comment]
I believe I know what is going on here. My attempt to change some of the dashboard inadvertently caused this. Fix should be in #1473.

**Comment by timurb** [comment]
One more update to the original issue: it seems the empty e-mail is send when error count in top right is increased. Not completely sure this is so but I'm sure I've seen both cases happening together when there is some issue with connectivity to OpenTSDB or with accessing data.

This is probably already fixed by pull-request but I feel I need to update the issue until the new version of Bosun holding the fix is out.
